NitroxServer-Subnautica/Serialization/ResourceAssets/ResourceAssetsHelper.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using AssetsTools.NET;
+using AssetsTools.NET.Extra;
+using NitroxModel.Logger;
+using NitroxServer.UnityStubs;
+
+namespace NitroxServer_Subnautica.Serialization.ResourceAssets
+{
+    public static class ResourceAssetsHelper
+    {
+        private static HashSet<ulong> matchingMonoscripts = new HashSet<ulong>();
+
+        public static AssetFileInfoEx FindComponentOfType<T>(AssetsFileInstance assetsFileInstance, ulong gameObjectPathId)
+        {
+            return FindComponentOfType(typeof(T), assetsFileInstance, gameObjectPathId);
+        }
+
+        public static AssetFileInfoEx FindTransform(AssetsFileInstance assetsFileInstance, ulong gameObjectPathId)
+        {
+            AssetsFileTable assetsTable = assetsFileInstance.table;
+            AssetsFile assetsFile = assetsFileInstance.file;
+            AssetsFileReader afr = assetsFile.reader;
+
+            ulong prevPos = assetsFile.reader.Position;
+            AssetFileInfoEx gameObjectAFI = assetsTable.getAssetInfo(gameObjectPathId);
+            if (gameObjectAFI.curFileType != 1)
+            {
+                Log.Info("An incorrect pathId was passed to FindTransform please check if it points to a proper gameObject: " + gameObjectPathId);
+                return null;
+            }
+
+            afr.Position = gameObjectAFI.absoluteFilePos;
+            afr.Align();
+
+            uint componentCount = afr.ReadUInt32();
+            AssetFileInfoEx componentAFI;
+
+            for (int i = 0; i < componentCount; i++)
+            {
+                uint fileId = (uint)afr.ReadInt32();
+                ulong pathId = (ulong)afr.ReadInt64();
+                if (fileId == 0)
+                {
+                    componentAFI = assetsTable.getAssetInfo(pathId);
+                    if (componentAFI.curFileType == 4) // Ey look we got it
+                    {
+                        afr.Position = prevPos;
+                        return componentAFI;
+                    }
+                }
+                else
+                {
+                    AssetsFileInstance dep = assetsFileInstance.dependencies[(int)fileId - 1];
+                    componentAFI = dep.table.getAssetInfo(pathId);
+                    if (componentAFI.curFileType == 4) // Ey look we got it
+                    {
+                        afr.Position = prevPos;
+                        return componentAFI;
+                    }
+                }
+            }
+
+
+
+            return null; // This should likely never happen
```

Comment:
Should we throw an exception with debugging information?  It looks like the callers are not checking for null.